jupyter/jupyter.go

```diff
@@ -240,6 +268,43 @@ func (s *server) iopubHandler(c *conn) {
 	}
 }
 
+func (s *server) heartbeatHandler(c *conn) {
+	if debug {
+		log.Printf("heartbeat handler started\n")
+	}
+
+	ctx, cancel := context.WithCancel(s.ctx)
```

Comment:
Is this cancellation context doing anything?

Reply:
not yet. (sorry, I forgot it when I crafted the CL)
but I have a CL pending that will call it when the kernel receives `"shutdown_request"`.